test/sequential/test-child-process-fork-getconnections.js

```diff
@@ -1,48 +1,50 @@
 'use strict';
-var assert = require('assert');
-var common = require('../common');
-var fork = require('child_process').fork;
-var net = require('net');
-var count = 12;
+const assert = require('assert');
+const common = require('../common');
+const fork = require('child_process').fork;
+const net = require('net');
+const count = 12;
 
 if (process.argv[2] === 'child') {
-  var sockets = [];
-  var id = process.argv[3];
+  let sockets = [];
 
   process.on('message', function(m, socket) {
+    function sendClosed(id) {
+      process.send({ id: id, status: 'closed'});
+    };
+
     if (m.cmd === 'new') {
       assert(socket);
       assert(socket instanceof net.Socket, 'should be a net.Socket');
       sockets.push(socket);
-      socket.on('end', function() {
-        if (!this.closingOnPurpose)
-          throw new Error('[c] closing by accident!');
-      });
     }
 
     if (m.cmd === 'close') {
       assert.equal(socket, undefined);
-      sockets[m.id].once('close', function() {
-        process.send({ id: m.id, status: 'closed' });
-      });
-      sockets[m.id].destroy();
+      if (sockets[m.id].destroyed) {
+        console.warn('socket already destroyed');
```

Comment:
In general, I agree, although most tests already write to console, including this one. I would make an exception for this particular item because this is the workaround for a bug elsewhere (and should be removed when the bug is fixed). I should add a comment to make that clear. So, I'll go do that now.